### PR TITLE
Adding REST service call recipe

### PR DESCRIPTION
**Pull request by ccoenraets**

### CODE REVIEW

File: force-app/main/default/lwc/miscRestApiCall/__tests__/miscRestApiCall.test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        document.body.appendChild(element);
+
+        // Validating that no p elements are rendered per default.
+        const detailEls = element.shadowRoot.querySelectorAll('p');
+        expect(detailEls.length).toBe(0);
+    });
+
+    it('renders book details based on a user query', () => {
+        const USER_INPUT = 'Harry Potter';
+        const BOOK_TITLES = Array.from(FETCH_DATA.items).map(
```
Comment (kevinv11n) [inline force-app/main/default/lwc/miscRestApiCall/__tests__/miscRestApiCall.test.js]: Why the need for `Array.from()`?

File: force-app/main/default/lwc/miscRestApiCall/miscRestApiCall.js [hunk 2]
```diff
@@ -17,12 +17,11 @@ export default class MiscRestCall extends LightningElement {
         fetch(QUERY_URL + this.searchKey)
```
[the remark this answers is not in the document]
Comment (jbleyleSF) [reply force-app/main/default/lwc/miscRestApiCall/miscRestApiCall.js]: We don't call this out in the LWC Dev Guide. Should we?

File: force-app/main/default/lwc/miscRestApiCall/__tests__/miscRestApiCall.test.js (hunk 1 above)
Comment (muenzpraeger) [reply force-app/main/default/lwc/miscRestApiCall/__tests__/miscRestApiCall.test.js]: Good catch!

File: force-app/main/default/lwc/miscRestApiCall/miscRestApiCall.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (muenzpraeger) [reply force-app/main/default/lwc/miscRestApiCall/miscRestApiCall.js]: I added a reference to the Fetch API in the recipe's template, and pointed out in the JS source that `fetch` doesn't work in IE11. IMHO that's good. Also it showcases how to properly work with `fetch` error handling. ;-)

**Pull request merged**